algorithm/cgal/cgal.cc

```diff
@@ -4974,47 +5206,48 @@ int Join(Geometry* geometry, int targets) {
 int Link(Geometry* geometry, bool close) {
   size_t size = geometry->size();
 
+  geometry->copyInputSegmentsToOutputSegments();
+  geometry->copyInputPointsToOutputPoints();
   geometry->transformToAbsoluteFrame();
 
   int target = geometry->add(GEOMETRY_SEGMENTS);
-  geometry->setTransform(target, new Transformation(CGAL::IDENTITY));
-  std::vector<Segment>& segments = geometry->segments(target);
+  geometry->copyTransform(target, Transformation(CGAL::IDENTITY));
+  std::vector<Segment>& out = geometry->segments(target);
 
   bool has_last = false;
   Point last;
 
   for (int nth = 0; nth < size; nth++) {
     switch (geometry->getType(nth)) {
       case GEOMETRY_SEGMENTS: {
-        std::vector<Segment>& input_segments = geometry->input_segments(nth);
-        if (input_segments.size() == 0) {
+        std::vector<Segment>& segments = geometry->segments(nth);
+        if (segments.empty()) {
           continue;
         }
         if (has_last) {
-          segments.emplace_back(last, input_segments[0].source());
+          out.emplace_back(last, segments.front().source());
         }
-        for (const Segment& segment : input_segments) {
-          segments.push_back(segment);
+        for (const Segment& segment : segments) {
+          out.push_back(segment);
         }
         has_last = true;
-        last = input_segments.end()[-1].target();
+        last = segments.back().target();
         break;
       }
       case GEOMETRY_POINTS: {
         // A point is equivalent to a zero-length segment.
```

Comment:
Is that really true? I know it's true mathematically, but is the representation of a point done by a zero length segment?

Reply:
Ah, a segment is represented as a pair of points -- source() and target().
A point in this case is a boring x, y, z.

Here we track the last point so that we can form a segment from it to the next point added.

Reply:
Gotcha, that's what I thought. I was just curious if there was some advantage to using the same representation for segments and for points.